packages/grafana-ui/src/components/Segment/SegmentInput.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useRef, useState } from 'react';
+import { cx, css } from 'emotion';
+import useClickAway from 'react-use/lib/useClickAway';
+import { useExpandableLabel, SegmentProps } from '.';
+
+export interface SegmentInputProps<T> extends SegmentProps<T> {
+  value: string | number;
+  onChange: (text: string | number) => void;
+}
+
+const textWidth = (text: string) => {
+  const element = document.createElement('canvas');
+  const context = element.getContext('2d')!;
+  context.font = '14px Roboto';
+  return context.measureText(text).width;
+};
```

Comment:
we have a util function for this: https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/utils/measureText.ts#L4 

has a cache as well to reuse canvas element